standards/application/messaging-api.md

```diff
@@ -0,0 +1,944 @@
+Content-Type: application/json
+
+{
+  "error": "Message with requestId 'xyz789' not found"
+}
+```
+
+##### By hash
+Request:
+```http
+GET /message?hash=abc123 HTTP/1.1
+Accept: application/json
+```
+
+Response:
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "sending": false,
+  "sent": false,
+  "stored": true,
+  "received": false,
+  "message": {
+    "payload": "SGVsbG8gd29ybGQ=",
+    "contentTopic": "/example/topic"
+  }
+}
+```
+
+Error for `404`:
+```http
+HTTP/1.1 404 Not Found
+Content-Type: application/json
+
+{
+  "error": "Message with hash 'abc123' not found"
+}
+```
+
+##### By `contentTopic`
+Request:
+```http
+GET /messages?contentTopic=/messaging/2/api/utf8&skip=0&take=10 HTTP/1.1
+Accept: application/json
+```
+Note: If skip and take are omitted, the request URL would be:
+```http
+GET /messages?contentTopic=/messaging/2/api/utf8 HTTP/1.1
+```
+
+Response:
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+[
+  {
+    "sent": false,
+    "stored": false,
+    "received": true,
+    "message": {
+      "payload": "SW5pdGlhbCBtZXNzYWdl",
+      "contentTopic": "/messaging/2/api/utf8"
+    }
+  }
+]
+```
+
+Error for `404`:
+```http
+HTTP/1.1 404 Not Found
+Content-Type: application/json
+
+{
+  "error": "No messages found for contentTopic '/messaging/2/api/utf8'"
+}
+```
+
+##### By `subscriptionId`
+Request:
+```http
+GET /messages?subscriptionId=1692854c-cbde-4afe-901f-1c86ecbd9ea1&skip=0&take=10 HTTP/1.1
+Accept: application/json
+```
+Note: When omitting pagination parameters, the URL would be:
+```http
+GET /messages?subscriptionId=1692854c-cbde-4afe-901f-1c86ecbd9ea1 HTTP/1.1
+```
+
+Response:
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+[
+  {
+    "sent": false,
+    "stored": false,
+    "received": true,
+    "message": {
+      "payload": "SW5pdGlhbCBtZXNzYWdl",
+      "contentTopic": "/messaging/2/api/utf8"
+    }
+  }
+]
+```
+
+Error for `404`:
+```http
+HTTP/1.1 404 Not Found
+Content-Type: application/json
+
+{
+  "error": "No messages found for subscriptionId '1692854c-cbde-4afe-901f-1c86ecbd9ea1'"
+}
+```
+
+### Health Indicator
+
+The `Health Indicator API` SHOULD be used to monitor the operational health of a node operating under the `Messaging API`.
+The criteria for determining a node’s health state MAY vary depending on the protocols in use and the node’s configuration.
+Detailed criteria for each health state are provided in the subsequent sections.
+
+For the purposes of this specification, three health states are defined:
+- `unhealthy`: Indicates that the node has lost connectivity for message reception, sending, or both, and as a result, it cannot reliably process or transmit messages.
+- `minimally healthy`: Indicates that the node meets the minimum operational requirements, although performance or reliability may be impacted.
+- `healthy`: Indicates that the node is operating optimally, with full support for message processing and transmission.
```

Comment:
delete that and just referred to the verbose definition in eponymous sections below